pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -523,6 +526,16 @@ def _process_response(self, response: chat.ChatCompletion | str) -> ModelRespons
         if not isinstance(response, chat.ChatCompletion):
             raise UnexpectedModelBehavior('Invalid response from OpenAI chat completions endpoint, expected JSON data')
 
+        # Some OpenAI-compatible providers (currently only Zhipu/Z.ai, see issue #2723) omit the `object` field even
+        # though the OpenAI schema includes it. We only patch it for that provider to avoid changing validation
+        # error counts in tests that purposefully feed invalid OpenAI responses (which expect 4 errors, including
+        # a missing `object`).
+        if self._provider.name == 'zhipu' and not getattr(response, 'object', None):  # pragma: no branch
+            try:  # defensive, in case attribute is read-only in future SDK versions
```

Comment:
Not necessary, it's a Pydantic `BaseModel` so fields should always be writable. If that stops being the case, we can fix it when we bump the SDK.